### PR TITLE
Add rabpro

### PR DESCRIPTION
**Pull request by jsta**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jsta** [comment]
I am willing to be listed as a maintainer.

**Comment by jonschwenk** [comment]
I am willing to be listed as a maintainer.

**Comment by jsta** [comment]
@conda-forge-admin ping conda-forge/help-python please.

File: recipes/rabpro/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+build:
+  noarch: python
+  number: 0  
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:  
+  host:
+    - pip
+    - python >=3.7,<3.10
```
Comment (carterbox) [inline recipes/rabpro/meta.yaml:22]:
```suggestion
    - python >=3.7
```

File: recipes/rabpro/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    
+  
+  run:
+    - python >=3.7,<3.10
```
Comment (carterbox) [inline recipes/rabpro/meta.yaml:26]:
```suggestion
    - python >=3.7
```
Looks like upstream only has a minimum python version; are you sure that this package isn't forward compatible? If you are afraid of python=4, you can use pin_compatible to the major version.

https://github.com/VeinsOfTheEarth/rabpro/blob/main/setup.py

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/rabpro/meta.yaml (hunk 2 above)
Comment (carterbox) [reply recipes/rabpro/meta.yaml:26]: If `fiona` is incompatible with python 3.11, then that is something that the `fiona` recipe will enforce. This recipe should only describe the requirements of rabpro.

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by jsta** [comment]
@conda-forge-admin, please restart ci

**Review by carterbox — APPROVED** [review]
Approved, but waiting for Windows build check to finish.

**Pull request merged**